esp/templates/survey/review.html

```diff
@@ -11,41 +11,131 @@
+    <ul style="list-style-type:circle;">
     {% for q in s.display_data.questions %}
+        {% if forloop.first %}
+            <li><a href="#survey{{ s.id }}-individual">Individual Questions</a></li>
+        {% endif %}
+    {% endfor %}
+    {% for c in s.perclass_data %}
+        {% if forloop.first %}
+            <li><a href="#survey{{ s.id }}-perclass">Per Class Questions</a></li>
+        {% endif %}
+    {% endfor %}
+    </ul>
+    </li>
+{% endfor %}
+
+{% if perclass_data %}
+{% for c in perclass_data %}
+    {% ifchanged c.class.parent_class %}
+        {% if not forloop.first %}
+            </ul>
+        {% endif %}
+        <li><a href="#class{{ c.class.parent_class.id }}">Responses for {{ c.class.parent_class.emailcode }}: {{ c.class.title }}</a></li>
+        <ul style="list-style-type:circle;">
+    {% endifchanged %}
+    <li><a href="#class{{ c.class.parent_class.id }}-{{ c.class.index }}">Responses for Section {{ c.class.index }}</a></li>
+    {% if forloop.last %}
+        </ul>
+    {% endif %}
+{% endfor %}
+{% endif %}
+</ul>
+
+<center>
+{% for s in surveys %}
+    <h1><a name="survey{{ s.id }}"></a>Responses for: {{ s.name }}<span style="float:right;font-weight: normal;"><a href="#top">[to top]</a></span></h1>
+    {% for q in s.display_data.questions %}
+        {% if forloop.first %}
+            <h2><a name="survey{{ s.id }}-individual"></a>Individual Questions<span style="float:right;font-weight: normal;"><a href="#top">[to top]</a></span></h2>
+            <table class="fullwidth bordered">
+        {% endif %}
         {% with s.num_participants as num_participants %}
-            {% include "survey/answers.html" %}
+            <tr><td>{% include "survey/answers.html" %}</td></tr>
         {% endwith %}
     {% endfor %}
+    </table></br></br>
+    {% for c in s.perclass_data %}
+        {% if forloop.first %}
+        <h2><a name="survey{{ s.id }}-perclass"></a>Per Class Questions<span style="float:right;font-weight: normal;"><a href="#top">[to top]</a></span></h2>
+        {% endif %}
+        {% ifchanged c.class.parent_class %}
+        {% if not forloop.first %}
+        </table></br></br>
+        {% endif %}
+        <table class="fullwidth bordered">
+        <tr><th><h2><a name="class{{ c.class.parent_class.id }}"></a>Responses for {{ c.class.parent_class.emailcode }}: {{ c.class.title }}<span style="float:right;font-weight: normal;"><a href="#top">[to top]</a></span></h2></th></tr>
+        {% endifchanged %}
+        <tr style="border-top: 1px dashed black;"><th><h3><a name="class{{ c.class.parent_class.id }}-{{ c.class.index }}"></a>Responses for Section {{ c.class.index }}<span style="float:right;font-weight: normal;"><a href="#top">[to top]</a></span></h3></th></tr>
+        {% for q in c.questions %}
+            {% with c.class.num_students as num_participants %}
+                <tr><td>{% include "survey/answers.html" %}</td></tr>
```

Comment:
For the per class questions in the html template, I'm only getting the "statistics:  Number of responses __/__" line for the first out of the per class questions for each class. In the pdf version, i get this statistic for each individual question.

Reply:
As the code currently stands, only countable questions will have the "Statistics" section included. Are your other questions short/long answer?

Reply:
OK, we now show the number of responses for every question, but I've moved it to next to the title (and now the other statistics are only shown if the question is numeric).